### PR TITLE
test(dm): fix unstable DM_RESYNC_NOT_FLUSHED_CASE

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: ref #4159

### What is changed and how it works?
check redirect and increase incremental data

#### Tests 

 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by lance6716** [comment]
opened https://github.com/pingcap/tiflow/issues/6603

File: dm/tests/shardddl4_1/run.sh [hunk 1]
```diff
@@ -1189,7 +1202,7 @@ function DM_RESYNC_NOT_FLUSHED_CASE() {
 	run_sql_source2 "insert into ${shardddl1}.${tb1} values(35,35,35);"
 	run_sql_source2 "insert into ${shardddl1}.${tb2} values(36,36,36);"
 
-	for ((k = 200; k < 240; k++)); do
+	for ((k = 200; k < 280; k++)); do
```
Comment (lance6716) [inline dm/tests/shardddl4_1/run.sh:1205]: can we reduce the DMLs because we now wait the status by checking the log?

**Comment by lance6716** [comment]
copied it into https://github.com/pingcap/tiflow/pull/6578 ;-)

File: dm/tests/shardddl4_1/run.sh (hunk 1 above)
Comment (lance6716) [reply dm/tests/shardddl4_1/run.sh:1205]: I tested it in https://github.com/pingcap/tiflow/pull/6578 , seems we don't need this change

**Review by lance6716 — APPROVED** [review]

**Review by buchuitoudegou — APPROVED** [review]

**Comment by lance6716** [comment]
please check which release version this PR should be cherry-picked @GMHDBJD

**Pull request merged**